inline-caching/README.md

```diff
@@ -0,0 +1,247 @@
+
+There is a working code examples for how to setup an inline cache here.
+```text
+$ git clone git@github.com:gemfire/gemfire-examples.git
+```
+
+## Configure Access to GemFire Maven Repository
+
+The quickstart tutorial requires access to the VMware Commercial Maven Repository for the GemFire product jars. Please sign-up for access to repo at <https://commercial-repo.pivotal.io/register>.
+
+Create a `gradle.properties` file in this directory with the following information:
+
+```plain
+gemfireRepoUsername=<your commercial maven repository username>
+gemfireRepoPassword=<your commercial maven repository password>
+```
+
+These properties will be used in the `web` and `cache-loader` gradle projects. 
+
+We can now build our example with 
+```bash
+./gradlew build
+```
+
+## Database setup
+
+### Install Postgres
+You can use any database you want that has a JDBC driver for it. For this example, we will be using Postgres. 
+You can either download postgres from https://www.postgresql.org/download/ and install it manually or if on a Mac, use brew install.
+
+```bash
+brew install postgresql
+brew services start postgresql
+```
+
+### Configure Postgres
+
+Now that postgres is running, we will want to create some tables and a user. One of the easier ways to do this
+is to use the test database that postgres provides for benchmarking via `pgbench`.
+
+```bash
+pgbench -i -s 50 postgres
+```
+
+This will create a few tables for us that we can manipulate. We will also need a user that our GemFire client can use to access the database
+
+```bash
+psql postgres
+create user myuser with encrypted password 'mypass';
+grant all privileges on database postgres to myuser;
+grant all privileges on all tables in schema public to myuser;
+```
+
+## Configure GemFire
+
+Once GemFire is installed, you need to configure GemFire to use Postgres as a data source. 
+This involves starting a GemFire cluster with all of the necessary class files as well as creating regions to store cached data and the data source to be used.
+
+This `cache-loader` project contains an implementation of `AsyncEventListener` and `CacheLoader` interfaces. 
+GemFire's AsyncEventListener and CacheLoader are useful features that can help improve the performance and scalability of your application by allowing you to asynchronously process write events and read data within your GemFire cluster.
+
+### AsyncEventListener
+The AsyncEventListener interface provides a way for your application to send write events to your database asynchronously in a batch operation, without blocking the sender. 
+This can help improve the throughput and performance of your application by allowing it to process events in parallel, while the sender continues to send more write events.
+This also helps keep the data in GemFire in sync with what is in the database.  
+
+In order to simplify the SQL necessary to write data to Postgres, the [JOOQ library](https://www.jooq.org/) is used. 
+
+```java
```

Comment:
What is the context for this code snippet?  Is this in a class already? If so we should include that...otherwise, the user doesn't have any context as to where this goes or how to use it.

We also don't introduce the code snippet, we just say we're using JOOQ...but what is this code doing?  Why does it matter?